sentry-core/src/test/java/io/sentry/core/transport/HttpTransportTest.kt

```diff
@@ -297,6 +297,66 @@ class HttpTransportTest {
         assertTrue(transport.isRetryAfter("event"))
     }
 
+    @Test
+    fun `parse X-Sentry-Rate-Limit and ignore unknown categories`() {
+        val transport = fixture.getSUT()
+
+        whenever(fixture.connection.inputStream).thenThrow(IOException())
+        whenever(fixture.connection.getHeaderField(eq("X-Sentry-Rate-Limits")))
+            .thenReturn("60:default;wtf;error;security:organization")
+
+        val event = SentryEvent()
+
+        transport.send(event)
+        Thread.sleep(2000)
+        assertFalse(transport.isRetryAfter("wtf"))
+    }
+
+    @Test
+    fun `When all categories is set but expired, applies only for specific category`() {
+        val transport = fixture.getSUT()
+
+        whenever(fixture.connection.inputStream).thenThrow(IOException())
+        whenever(fixture.connection.getHeaderField(eq("X-Sentry-Rate-Limits")))
+            .thenReturn("1::key, 60:default;error;security:organization")
+
+        val event = SentryEvent()
+
+        transport.send(event)
+        Thread.sleep(2000)
+        assertTrue(transport.isRetryAfter("event"))
+    }
+
+    @Test
+    fun `When default category is present, apply for error category`() {
+        val transport = fixture.getSUT()
+
+        whenever(fixture.connection.inputStream).thenThrow(IOException())
+        whenever(fixture.connection.getHeaderField(eq("X-Sentry-Rate-Limits")))
+            .thenReturn("60:default;security:organization")
+
+        val event = SentryEvent()
+
+        transport.send(event)
+        Thread.sleep(2000)
+        assertTrue(transport.isRetryAfter("event"))
+    }
+
+    @Test
+    fun `When category has shorter rate limiting, do not apply new timestamp`() {
+        val transport = fixture.getSUT()
+
+        whenever(fixture.connection.inputStream).thenThrow(IOException())
+        whenever(fixture.connection.getHeaderField(eq("X-Sentry-Rate-Limits")))
+            .thenReturn("60:error:key, 1:error:organization")
```

Comment:
`h`: I think it would be nice to also test if a new timestamp is applied if the category has a longer rate-limiting. Something like this `"2:error:organization 3:error:key, 1:error:organization"` and then test that the rate limit is actually 3 seconds. Here a DateProvider with whom we could move the date 3 seconds would already be handy.

Reply:
true, but this test is being done indirectly by all of the other tests.
I don't mind adding an extra one, but we don't have the DateProvider right now, can do next.